[STEP3] 프로필 편집 기능

nylah-j opened the pull request:
- 프로필 화면 및 프로필 편집 화면을 구현했습니다.
- 기능 동작 모습은 [README.md](https://github.com/nylah-j/w2-signup-profile/tree/step3)에서 확인하실 수 있습니다!

nylah-j committed:
[Step3] IBOutlet - IBAction

nylah-j committed:
[Step3] 새로운 화면 전달 및 데이터 전달

nylah-j committed:
[STEP3] 이미지 피커 화면을 delegate가 종료하도록 수정

nylah-j committed:
[STEP3] 프로필 편집 기능 구현

nylah-j committed:
[STEP3] 프로필 편집 기능 완료

nylah-j committed:
[STEP3] README 편집

nylah-j committed:
[STEP3] README 편집시 발생한 conflict resolve

nylah-j committed:
[STEP3] 로그인 및 웹뷰 구현

nylah-j committed:
[docs]] README에 로그인 및 웹뷰 구현 내용 추가

timothy-podo commented on KakaoProfile/KakaoProfile/EditViewController.swift:
```diff
@@ -0,0 +1,109 @@
+    }
+    
+    func initPhotoPickerButton() {
+        selectImageButton.clipsToBounds = true
```
clipsToBounds 을 설정하신 이유가 있을까요?

nylah-j replied:
갤러리로 이동하는 버튼을 둥글게 만들기 위해서 사용했습니다..!
`clipsToBounds`를 사용하면 버튼 안의 카메라 아이콘 서브뷰도 같이 둥글게 되는 것으로 알고 있었는데, 생각해보니 카메라 아이콘이 버튼 뷰를 벗어날 일이 없을것으로 보이네요!
없어도 되는 코드가 맞을까요??

timothy-podo replied:
아뇨, 의도대로 잘 쓰셨어요~

다른팀은 잘 모르겠고, 페이지의 경우 아래 layer.cornerRadius 를 통해서 깎아냈으니 이걸 적용하는 걸 layer.masksToBounds 로 보통 설정해 주고 있는데요. 
이 두 프로퍼티의 차이에 대해서도 학습하시면 어떨까 해서 링크 살짝 남겨봅니다.
https://stackoverflow.com/questions/1177775/how-is-the-relation-between-uiviews-clipstobounds-and-calayers-maskstobounds
https://zeddios.tistory.com/37

nylah-j replied:
view 내부에 layer가 존재하고, view를 통해서 clipsToBounds 값을 바꾸면 내부적으로 layer.masksToBounds의 값이 바뀌는 군요!
티모시 덕분에 view와 layer의 관계에 대해서 알 수 있었습니다. 
자료 감사합니다!!

timothy-podo commented on KakaoProfile/KakaoProfile/EditViewController.swift:
```diff
@@ -0,0 +1,109 @@
+        selectImageButton.clipsToBounds = true
+        selectImageButton.layer.cornerRadius = 0.5 * selectImageButton.bounds.size.width
+        
+        let profileImage = UIImage(named: "good_lion.png")
```
이미지를 프로젝트에 번들로 직접넣으셔서 확장자까지 full name 을 쓰신거 같네요? 이미지는 프로젝트에 번들로 넣기보다 Assets 에 등록하는 방식이 좋을거 같아요. 프로젝트를 열어보시면 Assets 라는게 있는데 한 번 확인 해 보실 수 있을까요?

nylah-j replied:
넵넵 asset을 활용해보겠습니다! 피드백 감사합니다~

timothy-podo commented on KakaoProfile/KakaoProfile/EditViewController.swift:
```diff
@@ -0,0 +1,109 @@
+    }
+    
+    @IBAction func doneButtonTouched(_ sender: Any) {
+        guard let name = nameTextField.text,
+              let description = descriptionTextField.text,
+              let profileImage = profileImageView.image else {
+                    return
+              }
+        
+        delegate?.sendProfile(profileImage: profileImage, name: name, description: description)
```
옵셔널 바인딩과 관련하여 테리의 PR 에서 제가 코멘트 드린게 하나 있는데, https://github.com/kakao-ent-2022/w2-signup-profile/pull/5#discussion_r787267905, 시간 나시면 함 살펴보셔도 좋을거 같아요.

nylah-j replied:
옵셔널로 넘어가는 것이면 중간에 unwrapping할 필요가 없었네요..!
옵셔널을 전달받을 수 있도록 변경했습니다. 피드백 감사합니다!!

nylah-j committed:
[FIX] 코드 리뷰 반영 - asset 폴더 이용, delegate 프로토콜에서 옵셔널 사용

- 앱 리소스를 asset 폴더 이용하여 관리하도도록 수정
- delegate 프로토콜에서 뷰의 자료를 받아서 전달하는 경우 optional 사용하도록 수정

nylah-j committed:
Merge branch 'step3' of https://github.com/nylah-j/w2-signup-profile into step3

timothy-podo commented on KakaoProfile/KakaoProfile/Assets.xcassets/good_lion.imageset/Contents.json:
```diff
@@ -0,0 +1,23 @@
+    {
+      "filename" : "good_lion.png",
+      "idiom" : "universal",
+      "scale" : "1x"
```
혹시 1x, 2x, 3x 에 이미지를 다르게 넣으신 이유가 있으실까요?
각 scale 에 넣으신 이미지가 사용되는 디바이스 해상도가 각각 어떻게 될까요?

nylah-j replied:
디바이스 해상도에 관계 없이 같은 이미지를 가져다가 사용한다는 생각에서 같은 이미지를 1x, 2x, 3x에 넣었습니다..!
다음부터는 해상도에 맞춰서 1x, 2x, 3x 를 업로드 하겠습니다!

kyoo-sl commented on .gitignore:
```diff
@@ -0,0 +1,15 @@
+*.xccheckout
+
+### Xcode Patch ###
+*.xcodeproj/*
```
@nylah-j  gitignore에 프로젝트 파일을 추가한 이유가 있을까요?
여러명이서 협업을 하면, xcodeproj도 팀별로 관리가 필요하거든요~

일단 gitignore에 프로젝트 파일을 추가하면, 예를 들어 제가 나일라의 코드를 받아도 엑스코드에서 나일라와 동일하게 실행되지 않아요~

nylah-j replied:
ignore 설정해주는 툴을 제가 잘 숙지하지 못하고 사용했습니다.
죄송합니다 😭  바로 gitignore 설정 반영했습니다!

kyoo-sl replied:
ㅎㅎ죄송할 일은 아니에요~
처음이니 그럴 수도 있죠!

nylah-j committed:
[FIX] gitignore에서 .xcodeproj 파일 제거

nylah-j committed:
[FIX] .xcodeproj 파일 업로드

bamiboo commented:
git으로 받아보니까 일부 파일들이 확인되지 않는데
혹시 파일생성당시 project.pbxproj 파일이 커밋누락된게 아닐까요? 😲

bamiboo-han commented:
git 으로 받아보니
이전에 삭제된듯한 파일들이 프로젝트 네비게이터(=파일목록)에 잡히고
sendProfile 함수 파라미터 타입도 EditProfileDelegate 에 선언된 부분과 다르게 commit 된것 같습니다(?)
(중간에 커밋이 꼬인게 아닐지 ㅎㅎㅎ)

nylah-j commented:
바로 조치하겠습니다..!!!
번거롭게 해드려서 죄송합니다..!!
